task/pylint/0.1/pylint.yaml

```diff
@@ -37,9 +37,16 @@ spec:
   steps:
     - name: lint
       image: docker.io/python:$(inputs.params.PYTHON)
-      workingDir: /workspace/source
+      workingDir: $(workspaces.source.path)
       script: |
         export PATH=$PATH:$HOME/.local/bin
-        pip install -r $(inputs.params.SOURCE_PATH)/$(inputs.params.REQUIREMENTS_FILE)
-        pip show pylint || echo "###\nWarning: Pylint is missing in your requirements\n###" && pip install pylint
+        if [ -n "$(inputs.params.REQUIREMENTS_FILE)" && -e "$(inputs.params.REQUIREMENTS_FILE)" ];then
```

Comment:
What does "-n" do? I've never seen that one before, and am having trouble finding it. https://tldp.org/LDP/abs/html/fto.html doesn't have it listed.

Reply:
checks if the var is defined  : 

```
$ [[ -n ${FOO} ]] && echo "yes"
chmouel@TheBatman:~
$ FOO=1
chmouel@TheBatman:~
$ [[ -n ${FOO} ]] && echo "yes"
yes
```

see man test : 

```     -n string     True if the length of string is nonzero.```